lib/mobilecommons.js

```diff
@@ -18,71 +17,63 @@ const auth = {
 };
 
 /**
-* Mobile Commons profile_update API. Can be used to subscribe the user to an
-* opt-in path.
-*
+* Mobile Commons profile_update API. Can be used to subscribe the user to an opt-in path.
 * @see https://mobilecommons.zendesk.com/hc/en-us/articles/202052534-REST-API#ProfileUpdate
 *
-* @param phone
-*   Phone number of the profile to update.
-* @param optInPathId
-*   Opt-in path to subscribe the user to.
-* @param customFields
-*   Object with MoCo custom profile field names as properties, and values to update the user with.
-*   Note: Field names are case-sensitive.
+* @param {object} profileId - Mobile Commons Profile ID
+* @param {number} phone - Phone to opt in
+* @param {string} oip - Opt-in Path to post to
+* @param {object} updateFields - field values to update on current User's Mobile Commons Profile
 */
-
-exports.profile_update = function(phone, optInPathId, customFields) {
+exports.profile_update = function(profileId, phone, oip, updateFields) {
+  const statName = 'mobilecommons: POST profile_update';
   if (process.env.MOBILECOMMONS_DISABLED) {
     logger.warn('MOBILECOMMONS_DISABLED');
 
     return;
   }
 
-  logger.log('debug', 'mobilecommons.profile_update for mobile:%s oip:%s customFields:%s', phone, optInPathId, JSON.stringify(customFields));
-
-  var url = 'https://secure.mcommons.com/api/profile_update';
-  var authEmail = process.env.MOBILECOMMONS_AUTH_EMAIL;
-  var authPass = process.env.MOBILECOMMONS_AUTH_PASS;
+  const updateTxt = JSON.stringify(updateFields);
+  // Use profileId if available to avoid logging.
+  let identifier = `profile:${profileId}`;
+  if (!profileId) {
+    identifier = `phone:${phone}`;
+  }
+  logger.info(`mobilecommons.profile_update ${identifier} oip:${oip} update:${updateTxt}`);
 
-  var postData = {
-    auth: {
-      user: authEmail,
-      pass: authPass
-    },
-    form:{
+  const data = {
+    auth,
+    form: {
       phone_number: phone,
-      opt_in_path_id: optInPathId
+      opt_in_path_id: oip
     }
   };
 
-  if (typeof customFields == 'object') {
-    var customFieldKeys = Object.keys(customFields);
-    for (var i = 0; i < customFieldKeys.length; i++) {
-      var key = customFieldKeys[i];
-      postData.form[key] = customFields[key];
-    }
+  if (typeof updateFields == 'object') {
```

Comment:
Right, it's already specified in the docblock. No need to check it additionally: if it's not an object it won't work correctly anyways.